lib/vdc/DerivedParticleDensity.cpp

```diff
@@ -0,0 +1,355 @@
+    vec3 diffu = maxu - minu;
+
+    vec3    binDimsF(xd, yd, zd);
+    ivec3   binDimsI(xd, yd, zd);
+    size_t  nbins = xd * yd * zd;
+    size_t *bins = new size_t[nbins];
+    double *accum = new double[nbins];
+    for (size_t i = 0; i < nbins; i++) {
+        bins[i] = 0;
+        accum[i] = 0;
+    }
+
+    auto it = inGrid->ConstCoordBegin();
+    auto dit = inGrid->cbegin();
+    auto end = inGrid->ConstCoordEnd();
+    for (size_t j = 0; it != end && j < realNP; ++it, ++dit, ++j) {
+        vec3  c((*it)[0], (*it)[1], (*it)[2]);
+        vec3  t = (c - minu) / diffu;
+        ivec3 i = t * binDimsF;
+        i = glm::clamp(i, ivec3(0), binDimsI - ivec3(1, 1, 1));
+        size_t index = i.z * yd * xd + i.y * xd + i.x;
+        bins[index]++;
+        accum[index] += *dit;
+    }
+
+    for (size_t i = 0; i < nbins; i++) output[i] = accum[i] / (float)bins[i];
+
+    delete[] accum;
+    delete[] bins;
+}
+
+
+// ===================================
+//        DerivedCoordVar1DSpan
+// ===================================
+
+
+DerivedCoordVar1DSpan::DerivedCoordVar1DSpan(string derivedVarName, DC *dc, string dimName, int axis, string units, float minExt, float maxExt)
+: DerivedCoordVar_CF1D(derivedVarName, dc, dimName, axis, units), _dc(dc), _minExt(minExt), _maxExt(maxExt)
+{
+    VAssert(maxExt >= minExt);
+}
+
+DerivedCoordVar1DSpan::DerivedCoordVar1DSpan(string derivedVarName, DC *dc, string dimName, int axis, string units, string inputCoordVar)
+: DerivedCoordVar1DSpan(derivedVarName, dc, dimName, axis, units, 0, 1)
+{
+    _inputCoordVar = inputCoordVar;
+}
+
+int DerivedCoordVar1DSpan::ReadRegion(int fd, const std::vector<size_t> &min, const std::vector<size_t> &max, float *region)
+{
+    VAssert(min.size() == 1);
+    VAssert(max.size() == 1);
+    long ts = -1;
+
+    if (!_derivedVarName.empty()) {
+        if (!_dc->IsCoordVar(_inputCoordVar)) {
+            assert(0);
+            return -1;
+        }
+
+        DC::CoordVar cvInfo;
+        _dc->GetCoordVarInfo(_inputCoordVar, cvInfo);
+        auto dims = cvInfo.GetDimNames();
+        if (dims.size() != 1) {
+            assert(0);
+            return -1;
+        }
+
+        DC::CoordVar myInfo;
+        GetCoordVarInfo(myInfo);
+        if (myInfo.GetAxis() != cvInfo.GetAxis()) {
+            assert(0);
+            return -1;
+        }
+
+        DC::FileTable::FileObject *f = _fileTable.GetEntry(fd);
+        if (!f) {
+            assert(0);
+            return -1;
+        }
+
+        DC::Dimension dim;
+        _dc->GetDimension(dims[0], dim, f->GetTS());
+        int fd = _dc->OpenVariableRead(f->GetTS(), _inputCoordVar);
+        if (fd < 0) {
+            assert(0);
+            return -1;
+        }
+        size_t nCoords = dim.GetLength();
+        float *coords = new float[nCoords];
+        ts = f->GetTS();
+
+        _dc->ReadRegion(fd, {0}, {nCoords - 1}, coords);
+        float min = FLT_MAX;
+        float max = -FLT_MAX;
```

Comment:
Will this work as anticipated? I.e. does it give the same result as std::numeric_limits<float>::lowest()?

Reply:
Yes.